README.md

```diff
@@ -1,2 +1,149 @@
+## Result example
+A result that can be generated for the usage example above:
+```json
+{
+  "sec-ch-ua-mobile": "?0",
+  "user-agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.72 Safari/537.36",
+  "accept-encoding": "gzip, deflate, br",
+  "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
+  "upgrade-insecure-requests": "1",
+  "accept-language": "en-US,en;0.9",
+  "sec-fetch-site": "same-site",
+  "sec-fetch-mode": "navigate",
+  "sec-fetch-user": "?1",
+  "sec-fetch-dest": "document"
+}
+```
+## API Reference
+All public classes, methods and their parameters can be inspected in this API reference.
+
+<a name="HeaderGenerator"></a>
+
+### HeaderGenerator
+HeaderGenerator randomly generates realistic browser headers based on specified options.
+
+
+* [HeaderGenerator](#HeaderGenerator)
+    * [`new HeaderGenerator(options)`](#new_HeaderGenerator_new)
+    * [`.getHeaders(options)`](#HeaderGenerator+getHeaders)
+
+
+* * *
+
+<a name="new_HeaderGenerator_new"></a>
+
+#### `new HeaderGenerator(options)`
+
+| Param | Type | Description |
+| --- | --- | --- |
+| options | [<code>HeaderGeneratorOptions</code>](#HeaderGeneratorOptions) | default header generation options used unless overridden |
+
+
+* * *
+
+<a name="HeaderGenerator+getHeaders"></a>
+
+#### `headerGenerator.getHeaders(options)`
+Generates a single set of headers using a combination of the default options specified in the constructor
+and their possible overrides provided here.
+
+
+| Param | Type | Description |
+| --- | --- | --- |
+| options | [<code>HeaderGeneratorOptions</code>](#HeaderGeneratorOptions) | specifies options that should be overridden for this one call |
+
+
+* * *
+
+<a name="prepareHttpBrowserObject"></a>
+
+### `prepareHttpBrowserObject(httpBrowserString)`
+
+| Param | Type | Description |
+| --- | --- | --- |
+| httpBrowserString | <code>string</code> | a string containing the browser name, version and http version, such as "chrome/88.0.4324.182|2" |
+
+
+* * *
+
+<a name="prepareBrowserObject"></a>
+
+### `prepareBrowserObject(browserString)`
+
+| Param | Type | Description |
+| --- | --- | --- |
+| browserString | <code>string</code> | a string containing the browser name and version, such as "chrome/88.0.4324.182" |
+
+
+* * *
+
+<a name="Browser"></a>
+
+### `Browser`
+
+| Param | Type | Description |
+| --- | --- | --- |
+| name | <code>string</code> | One of "chrome", "firefox" and "safari". |
+| minVersion | <code>number</code> | Minimal version of browser used. |
+| maxVersion | <code>number</code> | Maximal version of browser used. |
+| httpVersion | <code>string</code> | Either 1 or 2. If none specified the global `httpVersion` is used. |
```

Comment:
Shouldn't the `httpVersion` be determined automatically depending on the version of the browser?

Reply:
You can have http2 supported in the browser and not on the website, so you want to get HTTP1 headers even from HTT2 browser.

Reply:
@petrpatek @mnmkng, so how should this work? The interface is according to what Petr specified, but it can be changed, of course.

Reply:
Well, I don't know how it works, but I would expect it to work this way:

- no `httpVersion` configuration, default for browser is used.
- if local `httpVersion` is specified, it will be used.
- if global `httpVersion` is specified and local is not specified, that one will be used.

Importantly, there must not be any "default global", "default local" `httpVersion` which would override the natural HTTP version of the browser whose headers get generated. It can only be overridden explicitly by the user.

If it works like this, then it's fine, but the `Either 1 or 2. If none specified the global `httpVersion` is used.` does not explain this very well.

Reply:
Currently, there is a default global httpVersion of 2. Looking at it, I forgot to include the comment that mentioned it.
Anyway, it should be extremely easy to get rid of that default, so I will do that.

Reply:
@mnmkng I get your point. The problem is that the HTTP version is based on the website, not the browser. There is no default for the browser, but only default for the website. HTTP 2 is the default value because most modern websites are using http2. We cannot determine the HTTP version of the target website in this package. This will make the dev open the network activity and see the actual requests that the browser does. We could add this auto HTTP resolving to the got-scraping package - it should be easy with the HTTP resolver we already have in place.